source/acquaman/AMGenericStepScanController.cpp

```diff
@@ -71,88 +68,15 @@ void AMGenericStepScanController::onScanTimerUpdate()
 
 AMAction3 * AMGenericStepScanController::createInitializationActions()
 {
-	return 0;
+	return AMBeamline::bl()->createScanInitializationAction(configuration_);
 }
 
 AMAction3 * AMGenericStepScanController::createCleanupActions()
 {
-	return 0;
+	return AMBeamline::bl()->createScanCleanupAction(configuration_);
 }
 
 void AMGenericStepScanController::buildScanControllerImplementation()
 {
-	QList<AMXRFDetector *> xrfDetectors;
```

Comment:
Hmm... that's a tricky one.  I'll have to think about that.  @iainworkman @XuetaoLiu-CLS @davidChevrier feel free to provide suggestions as well.  More brains on the matter would be helpful.